staging/src/k8s.io/api/resource/v1alpha1/types.go

```diff
@@ -158,11 +158,57 @@ type AllocationResult struct {
 	// than one consumer at a time.
 	// +optional
 	Shareable bool `json:"shareable,omitempty" protobuf:"varint,3,opt,name=shareable"`
+
+	// Context is a backport of the field that replaces ResourceHandle in the
```

Comment:
Don't think of it as "a backport".  You're extending the v1alpha1 API.

What I am looking for in a change like this is: what happens if a v1alpha1 user sets BOTH fields?  It looks like you are asserting that `Context` is a superset of `ResourceHandle`, and `Context[0]` must match it.  At least, that's what the "internal" -> v1alpha1 seems to do. 

What happens if I get a v1alpha1 object with `ResourceHandle = "xyz"` and `Context = ["abc"]` ? 

Given that this is an alpha API, I'm willing to let things fly that wouldn't fly in GA, but you need to document what happens if I create such a monstrosity. Do we just blow away ResourceHandle and replace it with Context?  Or do we fail at validation saying "these must match" (which is what we do for service `clusterIP` and `clusterIPs`)?

What happens if a update an object from `resourceHandle: "xyz"`  `resourceHandle: "xyz", context: ["abc"]` ? Is that an error?  Worse, what if I update from `resourceHandle: "xyz"`  `resourceHandle: "def", context: ["abc"]` ?

Reply:
Now that we have decided to nuke `v1alpha1` completely, this is no longer relevant (but still good to know for the future).